sdk/eventhub/eventhubs-checkpointstore-table/src/tableCheckpointStore.ts

```diff
@@ -42,13 +109,69 @@ export class TableCheckpointStore implements CheckpointStore {
    *  - `tracingOptions`: Options for configuring tracing.
    * @returns A list partitions this instance successfully claimed ownership.
    */
-  async claimOwnership(): /*
-    partitionOwnership: PartitionOwnership[],
-    options: OperationOptions = {}
-    */
-  Promise<PartitionOwnership[]> {
-    console.log("nothing to claim");
-    throw new Error("not implemented");
+  async claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]> {
+    const partitionOwnershipArray: PartitionOwnership[] = [];
+
+    for (const ownership of partitionOwnership) {
+      const partition_Key = `${ownership.fullyQualifiedNamespace} ${ownership.eventHubName} ${ownership.consumerGroup} Ownership`;
+      const ownershipEntity: PartitionOwnershipEntity = {
+        partitionKey: partition_Key,
+        rowKey: ownership.partitionId,
+        ownerid: ownership.ownerId
+      };
+
+      // When we have an etag, we know the entity existed.
+      // If we encounter an error we should fail.
+      try {
+        if (ownership.etag) {
+          const updatedMetadata = await this._tableClient.updateEntity(ownershipEntity, "Replace", {
+            etag: ownership.etag
+          });
+          ownership.etag = updatedMetadata.etag;
+          ownership.lastModifiedTimeInMs = await this._toMillisecs(
+            (updatedMetadata as any).Timestamp
+          );
+          partitionOwnershipArray.push(ownership);
+          logger.info(
+            `[${ownership.ownerId}] Claimed ownership successfully for partition: ${ownership.partitionId}`,
+            `LastModifiedTime: ${ownership.lastModifiedTimeInMs}, ETag: ${ownership.etag}`
+          );
+        } else {
+          try {
```

Comment:
You already have a `try` wrapping the entire `if/else` blocks, so is this 2nd `try` really needed?